src/Tools/dotnet-monitor/GenerateApiKeyCommandHandler.cs

```diff
@@ -18,39 +22,84 @@ namespace Microsoft.Diagnostics.Tools.Monitor
     /// </summary>
     internal sealed class GenerateApiKeyCommandHandler
     {
-        public Task<int> GenerateApiKey(CancellationToken token, int keyLength, string hashAlgorithm, IConsole console)
+        public Task<int> GenerateApiKey(CancellationToken token, OutputFormat output, IConsole console)
         {
-            if (!HashAlgorithmChecker.IsAllowedAlgorithm(hashAlgorithm))
+            GeneratedJwtKey newJwt = GeneratedJwtKey.Create();
+
+            StringBuilder outputBldr = new StringBuilder();
+
+            outputBldr.AppendLine(FormattableString.Invariant($"{Strings.Message_GenerateApiKey}"));
+            outputBldr.AppendLine();
+            outputBldr.AppendLine(FormattableString.Invariant($"{HeaderNames.Authorization}: {AuthConstants.ApiKeySchema} {newJwt.Token}"));
+            outputBldr.AppendLine();
+
+            RootOptions opts = new RootOptions()
             {
-                console.Error.WriteLine(
-                    string.Format(
-                        CultureInfo.CurrentCulture, 
-                        Strings.ErrorMessage_ParameterNotAllowed, 
-                        nameof(hashAlgorithm), 
-                        hashAlgorithm));
-                return Task.FromResult(1);
-            }
+                Authentication = new AuthenticationOptions()
+                {
+                    MonitorApiKey = new MonitorApiKeyOptions()
+                    {
+                        Subject = newJwt.Subject,
+                        PublicKey = newJwt.PublicKey,
+                    }
+                }
+            };
 
-            if (keyLength < ApiKeyAuthenticationHandler.ApiKeyByteMinLength || keyLength > ApiKeyAuthenticationHandler.ApiKeyByteMaxLength)
+            outputBldr.AppendFormat(CultureInfo.CurrentCulture, Strings.Message_SettingsDump, output);
+            outputBldr.AppendLine();
+            switch (output)
             {
-                console.Error.WriteLine(
-                    string.Format(
-                        CultureInfo.CurrentCulture,
-                        Strings.ErrorMessage_ParameterNotAllowedByteRange,
-                        nameof(hashAlgorithm),
-                        hashAlgorithm,
-                        ApiKeyAuthenticationHandler.ApiKeyByteMinLength,
-                        ApiKeyAuthenticationHandler.ApiKeyByteMaxLength));
-                return Task.FromResult(1);
+                case OutputFormat.Json:
+#if NET5_0_OR_GREATER
```

Comment:
I've now seen variations of this at least three times. I'm thinking it might be goo to write an extension method for JsonSerializerOptions (e.g. `public static JsonSerializerOptions SkipWritingNull(this JsonSerializerOptions options)`) so then we don't have to condition the compilation and it might be more concise, e.g.:

```cs
JsonSerializer.Serialize(options, new JsonSerializerOptions() { WriteIndented = true }.SkipWritingNull());
```

or maybe a static method that just creates a new instance of JsonSerializerOptions with ignoring null by default e.g.

```cs
JsonSerializerOptions jsonOptions = CreateSkipWritingNull();
jsonOptions .WriteIndented = true;
JsonSerializer.Serialize(options, jsonOptions);
```

Basically, any solution that avoids scattering #if NET5_0_OR_GREATER everywhere for this is better.